add  AWS STS temporary credentials for Athena (aws_session_token) (#8822)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/connect-data-platform/connect-amazon-athena.md

| AWS Secret Access Key | aws_secret_access_key | Secret access key of the user performing requests                          | String | Required | wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY |
| Schema                | schema                | Specify the schema (Athena database) to build models into (lowercase only) | String | Required | dbt |
| Threads               | threads               |                                                                            | Integer| Optional | 3 |

### Temporary credentials

> If you prefer to not store long-lived IAM user AWS Access Key ID in the <Constant name="dbt_platform" /> user interface (UI), you can use `aws_session_token`, a [AWS security token service (STS) temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html), instead. This way, access mirrors a user's long-term access key credentials (user or role).

Reviewer: ```suggestion
If you prefer to not store long-lived IAM user AWS Access Key ID and AWS Secret Access Key in the <Constant name="dbt_platform" />, you can use `aws_session_token`, which is part of [temporary AWS Security Token Service (STS) credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html), instead. This approach mirrors a user or role’s long-term permissions.
```

> If you choose this route, make sure you enter:

Reviewer: ```suggestion
To use temporary credentials:
```

> - The AWS Access Key ID and AWS Secret Access Key in the [development environment settings](/docs/dbt-cloud-environments#create-a-development-environment) in the <Constant name="dbt_platform" /> user interface (UI).

Reviewer: Should users do this in order? If so, I suggest we use numbered steps:
```suggestion
1. Enter the **AWS Access Key ID** and **AWS Secret Access Key** in the [development environment settings](/docs/dbt-cloud-environments#create-a-development-environment) in the <Constant name="dbt_platform" /> user interface (UI).
```

> - Since the `aws_session_token` isn't available as a UI field in <Constant name="dbt_platform" />, you can add it using [Extended attributes](/docs/dbt-cloud-environments#extended-attributes) in the environment settings or [<Constant name="dbt_platform" /> profiles](/docs/cloud/about-profiles). 

Reviewer: ```suggestion
2. Since the `aws_session_token` isn't available as a UI field in <Constant name="dbt_platform" />, add it using [Extended attributes](/docs/dbt-cloud-environments#extended-attributes) in the environment settings or [<Constant name="dbt_platform" /> profiles](/docs/cloud/about-profiles). 
```

>   When you set the `aws_session_token` in **Extended attributes**, the key is case-sensitive and must be exactly `aws_session_token`.

Reviewer: It looks like lines 49-55 need to be indented more:

<img width="903" height="273" alt="Image" src="https://github.com/user-attachments/assets/fd7a3ba2-37c1-4536-a989-b7c0b460a888" />

> When you set the `aws_session_token` in Extended Attributes, the key is case-sensitive and must be exactly `aws_session_token`.

Reviewer: If we're using numbered steps, lines 49-55 should be under Step 2. 
```suggestion
When you set the `aws_session_token` in **Extended attributes**, the key is case-sensitive and must be exactly `aws_session_token`.
```

For example, you can use an [environment variable](/docs/build/environment-variables) so the token isn't stored in the text box UI:

```yaml
aws_session_token: '{{ env_var(''DBT_ENV_AWS_SESSION_TOKEN'') }}'
```

> [Assign](/docs/build/environment-variables#setting-environment-variables) `DBT_ENV_AWS_SESSION_TOKEN` for each environment. Since temporary credentials expire, you need to refresh your STS credentials and update the environment (or the variable value) before expiration.

Reviewer: ```suggestion
3. [Assign](/docs/build/environment-variables#setting-environment-variables) `DBT_ENV_AWS_SESSION_TOKEN` for each environment. Since temporary credentials expire, you need to refresh your STS credentials and update the environment (or the variable value) before expiration.
```
